src/main/java/com/transferwise/openbanking/client/api/common/ApiResponse.java

```diff
@@ -0,0 +1,83 @@
+package com.transferwise.openbanking.client.api.common;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/**
+ * A container for the response to an API call to an ASPSP, which may be a successful response or a failure response.
+ *
+ * @param <T> The type returned in the response body, when the API call is successful
+ * @param <U> The type returned in the response body, when the API call is un-successful, and a structured error
+ *            response is returned
+ */
+@Getter
+@AllArgsConstructor
+public class ApiResponse<T, U> {
+
+    private static final int CLIENT_ERROR_STATUS_START_RANGE = 400;
```

Comment:
Intentionally not using Spring HTTP client libraries here to make the library in general less dependent on Spring.